components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/internal/RoleManagementServiceImpl.java

```diff
@@ -47,14 +47,14 @@ public RoleBasicInfo addRole(String roleName, List<String> userList, List<String
     }
 
     @Override
-    public List<RoleBasicInfo> getRoles(int limit, int offset, String sortBy, String sortOrder, String tenantDomain)
-            throws IdentityRoleManagementException {
+    public List<RoleBasicInfo> getRoles(Integer limit, Integer offset, String sortBy, String sortOrder,
```

Comment:
any reason why we can't have primitives here?

Reply:
There can be null values passed from the SCIM side if the user does not provide those values. So we validate it here and assign default values.